File: src/color.ts
export class Color {

    red: number;
    green: number;
    blue: number;

    constructor(red: number, green: number, blue: number) {
        this.red = red;
        this.green = green;
        this.blue = blue;
    }

    scale(val: number) {
        this.red *= val;
        this.green *= val;
        this.blue *= val;
    }

    add(otherColor: Color) {
        this.red += otherColor.red;
        this.green += otherColor.green;
        this.blue += otherColor.blue;
    }

    times(otherColor: Color) {
        this.red *= otherColor.red;
        this.green *= otherColor.green;
        this.blue *= otherColor.blue;
    }

    clone(): Color {
        return new Color(this.red, this.green, this.blue);
    }

    set(color: Color) {
        this.red = color.red;
        this.green = color.green;
        this.blue = color.blue;
    }
}

File: color.go
package main

type Color struct {
	red   float64
	green float64
	blue  float64
}

func (color *Color) scale(val float64) {
	color.red *= val
	color.green *= val
	color.blue *= val
}

func (color *Color) add(otherColor Color) {
	color.red += otherColor.red
	color.green += otherColor.green
	color.blue += otherColor.blue
}

func (color *Color) times(otherColor Color) {
	color.red *= otherColor.red
	color.green *= otherColor.green
	color.blue *= otherColor.blue
}

func (color *Color) clone() Color {
	return Color{red: color.red, green: color.green, blue: color.blue}
}

func (color *Color) set(other Color) {
	color.red = other.red
	color.green = other.green
	color.blue = other.blue
}
